default/python/AI/AIFleetMission.py

```diff
@@ -62,6 +62,44 @@ def __init__(self, fleet_id):
         self.type = None
         self.target = None
 
+    def __setstate__(self, state):
+        assert type(state) == dict
+        assert len(state) == 5
```

Comment:
explain why 5 in a comment

Reply:
I suggest making a list of the named items expected to be in the state, and then comparing the length of the state to the length of that list.  That makes it a tad more clear what's going on, and makes any update to it a clear process also (and could also be used to assist an error message).